Display tooltip when NxTile.HeaderTitle and NxDrawer.HeaderTitle is truncated - RSC-1131

entroform opened the pull request:
https://issues.sonatype.org/browse/RSC-1131

I'm not 100% sure if a test is warranted for these additions since they are implementation details and `NxOverflowTooltip` is already tested, let me know your thoughts!

rpokorny commented on lib/src/components/NxDrawer/NxDrawer.tsx:
```diff
@@ -12,18 +12,44 @@ import React, {
   closeDrawer: () => {}
 });
 
+const NxDrawerHeaderTitle = (props: NxDrawerHeaderTitleProps) => {
```
The prior implementation was able to accept a ref. This one should too to avoid a breaking change

entroform replied:
> The prior implementation was able to accept a ref. This one should too to avoid a breaking change

Thanks! I forwarded refs here: febd40de14d6f870840366f6a992f2ed1f0bb41b

rpokorny commented on lib/src/components/NxTile/NxTile.tsx:
```diff
@@ -0,0 +1,45 @@
+
+import { NxTileHeaderTitleProps } from './types';
+
+const NxTileHeaderTitle = (props: NxTileHeaderTitleProps) => {
```
Accept a ref

rpokorny commented on lib/src/components/NxTile/types.ts:
```diff
@@ -0,0 +1,11 @@
+import { HTMLAttributes, ReactNode } from 'react';
+
+export interface NxTileHeaderTitleProps extends HTMLAttributes<HTMLDivElement> {
+  children: ReactNode;
```
Is this not already in `HTMLAttributes<HTMLDivElement>`?

entroform replied:
> Is this not already in `HTMLAttributes<HTMLDivElement>`?

Yup! I fixed that as part of: ebd40de14d6f870840366f6a992f2ed1f0bb41b
(Also updated the ones in NxDrawer)

rpokorny commented on lib/src/components/__tests__/SimpleComponents.test.tsx:
```diff
@@ -117,66 +116,6 @@ describe('NxH4', function() {
   });
 });
 
-describe('NxTile', function() {
```
These tests still need to exist, even if not in this file

entroform replied:
> These tests still need to exist, even if not in this file

Brought it back here: d024164426708062d22c527c0a562490c1d6a73a

rpokorny commented:
> I'm not 100% sure if a test is warranted for these additions since they are implementation details and NxOverflowTooltip is already tested, let me know your thoughts!

I'd say visual tests are needed.  We recently added a third parameter to `hoverTest` that allows it to wait for a tooltip; that would be good to use here.

entroform commented:
> > I'm not 100% sure if a test is warranted for these additions since they are implementation details and NxOverflowTooltip is already tested, let me know your thoughts!
> 
> I'd say visual tests are needed. We recently added a third parameter to `hoverTest` that allows it to wait for a tooltip; that would be good to use here.

Hi Ross, I updated the visual tests here: a54b9e73d7d8659f8cc7da02d5f48b941b0a9245
I had implemented it with hoverTest before but the tooltips we're cut off, let me know if this approach is okay.

rpokorny commented on gallery/src/styles/NxTile/NxTilesExamples.tsx:
```diff
@@ -152,6 +154,14 @@ const NxTilesExamples = () =>
       out as expected.
     </GalleryExampleTile>
+
+    <GalleryExampleTile title="Simple Convenience Components Example"
```
Needs `defaultCheckeredBackground={true}`.  This will make the visual test snapshot clearer too

entroform replied:
> Needs `defaultCheckeredBackground={true}`. This will make the visual test snapshot clearer too

I added that here a0e0084e4b69ed42779a44b25097c40538e19713
For some reason the snapshots didn't update or the checkered board is not showing in the visual tests. taking a look at that now

entroform replied:
> Needs `defaultCheckeredBackground={true}`. This will make the visual test snapshot clearer too

Updated snapshots here: 99e69dadf1767a2ef9d30cf8d02712388d4ee26f
